sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/randomExpressions.scala

```diff
@@ -102,6 +105,11 @@ case class Rand(child: Expression) extends RDG with ExpressionWithRandomSeed {
   }
 
   override def freshCopy(): Rand = Rand(child)
```

Comment:
What happens when we do `freshCopy`? Do we need to propagate `useRandSeed` field?

Reply:
oh, yes. Nice catch!